doc/pages/Manual.md

```diff
@@ -7,33 +7,228 @@ primarily of use for packagers, package maintainers and repository maintainers.
-  (OUTDATED)
 * If you want to hack on opam or build related tools, the API documentation can
   be browsed [here](api/index.html)
 
+## Base concepts
+
+### Opam root
+
+Opam holds its configuration, metadata, logs, temporary directories and caches
+within a directory that we will call _opam root_. By default, this is `~/.opam`,
+and we may refer to it by this name in this manual for the sake of simplicity,
+but this can be changed using the `OPAMROOT` environment variable or the
+`--root` command-line argument.
+
+An existing opam root is required for opam to operate normally, and one is
+created upon running `opam init`. The initial configuration can be defined
+through a configuration file at `~/.opamrc`, `/etc/opam` or at a location
+specified through the `--config` command-line option. If none is present, opam
+defaults to its built-in configuration that binds to the OCaml repository at
+`https://opam.ocaml.org`.
+
+Except in explicit cases, opam only alters files within its opam root. It is
+organised as follows:
+- [`~/.opam/config`](#config): the global opam configuration file
+- `~/.opam/repo/`: contains the mirrors of the configured package repositories
+- [`~/.opam/repo/repos-config`](#reposconfig): lists the configured package repositories and their URLs
+- `~/.opam/repo/<repo>`: mirror of the given repository
+- `~/.opam/opam-init/`: contains opam configuration scripts for the outside world, e.g. shell environment initialisation
+- `~/.opam/cache/`: caches of downloaded files
+- `~/.opam/plugins/<plugin>`: reserved for plugins
+- `~/.opam/<switch>`: prefixes of named [switches](#switches)
+
+### Repositories
+
+Repositories are collection of opam package definitions. They respect the
+following hierarchy:
+- [`/repo`](#repo): repository configuration file
+- [`/packages/<pkgname>/<pkgname>.<version>/opam`](#opam): holds the metadata
+  for the given package. `url` and `descr` may also be present, in which case
+  they override what may already be present in the `opam` file
+- [`/packages/<pkgname>/<pkgname>.<version>/files/`](#files): contains files to copy over
+  the source of the given package before it gets used
+- `/archives/<pkgname>.<version>+opam.tar.gz`: cached package archives
+  (as generated by running `opam-admin make`)
+- `/urls.txt`, `/index.tar.gz`: index files useful when serving over HTTP, they
+  can be generated using `opam-admin make --index`
+
+Opam can be configured to use repositories at local or remote (ssh) paths, HTTP
+URLs, or on version control systems. A repository is set up using
 
-This version of the manual documents the version `1.2` of the file formats, and
-is progressively being updated with the new features of opam `2.0`.
+```
+opam repository add <name> <URL>
+```
+
+and can subsequently be then selected for use in specific switches using `opam
+repository select <name>`. Use `opam repository list --all` for an overview of
```

Comment:
I don't like this two step procedure which forces you to introduce the `select` terminology (how do you handle remove `unselect` ?). The whole thing feels bureaucratic from an UI point of view. I think this could be simplified by having something like:

```
opam repo add [--switch SWITCH|--to-list] NAME [URL]
```
Basic semantics is:
```
opam repo add name            # add name to current switch (must exist in the global list)
opam repo add --switch s name # same as above but do this in switch s 
opam repo add name url        # add name/url to global list and to current switch (error if name exists)
opam repo add --to-list name url  # add name/url to global list only
etc.
```